x/stakeibc/keeper/reward_allocation.go

```diff
@@ -1,86 +1,32 @@
-	rewardedTokens := k.bankKeeper.GetAllBalances(ctx, rewardCollectorAddress)
-	if rewardedTokens.IsEqual(sdk.Coins{}) {
-		k.Logger(ctx).Info("No reward to allocate from RewardCollector")
-		return false
-	}
-
-	rewardsAccrued := false
-	for _, token := range rewardedTokens {
-		// get hostzone by reward token (in ibc denom format)
-		hz, err := k.GetHostZoneFromIBCDenom(ctx, token.Denom)
-		if err != nil {
-			k.Logger(ctx).Info("Token denom %s in module account is not from a supported host zone", token.Denom)
-			continue
-		}
-
-		// liquid stake all tokens
-		msg := types.NewMsgLiquidStake(rewardCollectorAddress.String(), token.Amount, hz.HostDenom)
-		if err := msg.ValidateBasic(); err != nil {
-			k.Logger(ctx).Error(fmt.Sprintf("Liquid stake from reward collector address failed validate basic: %s", err.Error()))
-			continue
-		}
-		_, err = msgSvr.LiquidStake(ctx, msg)
-		if err != nil {
-			k.Logger(ctx).Error(fmt.Sprintf("Failed to liquid stake %s for hostzone %s: %s", token.String(), hz.ChainId, err.Error()))
-			continue
-		}
-		k.Logger(ctx).Info(fmt.Sprintf("Liquid staked %s for hostzone %s's accrued rewards", token.String(), hz.ChainId))
-		rewardsAccrued = true
-	}
-	return rewardsAccrued
-}
+// AuctionOffRewardCollectorBalance transfers all balances from the reward collector module account
+// to the auction module account. If the reward collector has no balance, it does nothing.
+func (k Keeper) AuctionOffRewardCollectorBalance(ctx sdk.Context) {
+	k.Logger(ctx).Info("Auctioning reward collector balance")
 
-// Sweep stTokens from Reward Collector to Fee Collector
-func (k Keeper) SweepStTokensFromRewardCollToFeeColl(ctx sdk.Context) error {
-	// Send all stTokens to fee collector to distribute to delegator later
 	rewardCollectorAddress := k.AccountKeeper.GetModuleAccount(ctx, types.RewardCollectorName).GetAddress()
 
-	rewardCollCoins := k.bankKeeper.GetAllBalances(ctx, rewardCollectorAddress)
-	k.Logger(ctx).Info(fmt.Sprintf("Reward collector has %s", rewardCollCoins.String()))
-	stTokens := sdk.NewCoins()
-	for _, token := range rewardCollCoins {
-		// get hostzone by reward token (in stToken denom format)
-		isStToken := k.CheckIsStToken(ctx, token.Denom)
-		k.Logger(ctx).Info(fmt.Sprintf("%s is stToken: %t", token.String(), isStToken))
-		if isStToken {
-			stTokens = append(stTokens, token)
-		}
+	rewardCollectorBalances := k.bankKeeper.GetAllBalances(ctx, rewardCollectorAddress)
+	if rewardCollectorBalances.Empty() {
+		k.Logger(ctx).Info("No rewards to auction from RewardCollector")
+		return
 	}
-	k.Logger(ctx).Info(fmt.Sprintf("Sending %s stTokens from %s to %s", stTokens.String(), types.RewardCollectorName, authtypes.FeeCollectorName))
 
-	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.RewardCollectorName, authtypes.FeeCollectorName, stTokens)
+	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.RewardCollectorName, auctiontypes.ModuleName, rewardCollectorBalances)
```

Comment:
just flagging that we don't double check that there's an auction registered, but I think that's the right behavior!

Is the 2nd iteration of this when we plan to have separate module accounts for each auction?

Reply:
IMO the next iteration will have a function in x/auction that gets a selling token and returns the address of the auction that handles this selling token.